wordpress-mobile/WordPress-Android
# Enable new Site Creation flow

This PR updates the FluxC hash which includes a change to which localization parameter name is used for wpcom v2 endpoints. This change has been done in this FluxC PR: https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1110

That ^ change and some server side changes were the last blockers for the new site creation flow, so we are also enabling the feature flag for it for all builds. We'll keep this feature flag for now in case we need to revert the flow due to a major issue.

**To test new site creation**

New site creation flow has been tested numerous times on different devices and APIs, but we can never have enough tests.

You can access the new site creation flow by following the steps outlined below. There is nothing specific that needs to be tested as part of enabling the feature flag, so a test to verify that it works should be enough, but feel free to test it in as much detail as you'd like.

**Accessing new site creation flow**
* Go to the site picker page
* Tap on the + menu button
* Select "Create WordPress.com site"

**To test localization change**

We'll need to test 2 different things for localization. First, we need to ensure that the localization change didn't effect anything for wpcom v1 endpoints.

* Change the interface language from the "App Settings" under "Me" tab. Here is a gif of how to do it: https://cloudup.com/cCA56PJEaa5
* Do a general test of the "My Site" tab for a Wordpress.com site and verify that everything works as expected (as it used to)
* One easy test would be to go to the `Plans` page and check that the plans are correctly localized
* Don't forget that some things are cached (network requests, views etc), so sometimes you'll need to trigger a new network request or a view refresh to get the correctly localized version after a language change. One example of this would be the "Free" label in the `Plans` row of the My site page. If you see that it's not correctly localized, just go back to site picker and select the site again which should trigger an update.

Here is my test for the above steps: https://cloudup.com/cddkDJBzl4D

---

The second test we'll need to do is to check that the new site creation flow results are correctly localized. The endpoints this new flow uses are the only v2 endpoints that are actually localized, so we'll use that to verify the `_locale` parameter change is implemented correctly.

* Change the interface language from the "App Settings" under "Me" tab. Here is a gif of how to do it: https://cloudup.com/cCA56PJEaa5
* Follow the steps above to access the new site creation flow
* Verify that the segments, verticals and the header in the site info pages are localized. Note that not everything is translated right now, so ensuring that some of the results being localized should be enough.

_P.S: I found that sometimes I need to restart the app for the language changes to be reflected. I verified that we are still using the `_locale` parameter, but the old language is being sent. We have not made any changes to this in this or the FluxC PR, so it's not an issue with it. I'll try to figure out why it's happening, but it should not block this PR._

Bonus points for tests: Use stetho to check the network requests and ensure that the v1 endpoints have `locale=xx_XX` parameter and the v2 wpcom endpoints have the `_locale=xx_XX` parameter. Here is a screenshot from one of my tests: https://cloudup.com/cvI2HXyoFIf

Update release notes:


It looks like the release notes already has the `Introduced new site creation wizard` due to a merging issue which I reported to Platform 9: https://github.com/wordpress-mobile/WordPress-Android/blob/develop/RELEASE-NOTES.txt#L1 - That's why this PR is not adding that change.
